src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -398,6 +454,18 @@ public string BindValues(RouteValueDictionary acceptedValues)
 
         private bool TryBindValuesCore(UriBuildingContext context, RouteValueDictionary acceptedValues)
         {
+            // If we have any output parameter transformers, allow them a chance to influence the parameter values
+            // before we build the URI.
+            var parameterTransformers = _parameterTransformers;
+            for (var i = 0; i < parameterTransformers.Length; i++)
+            {
+                var parameterName = parameterTransformers[i].parameterName;
+                if (acceptedValues.TryGetValue(parameterName, out var value))
+                {
+                    acceptedValues[parameterName] = parameterTransformers[i].transformer.Transform(value);
```

Comment:
I started there, and it will require some rewriting. The existing code actually has a very subtle bug - because parameters don't get called when the value gets passed to `Buffer(...)`.

I need to log the bug regardless because I think lowercase is broken.

Reply:
https://github.com/aspnet/Routing/issues/802

Reply:
So what happens when you do this is that you have move `Convert.ToString` into accept and buffer.

We also have another problem in the current implementation where route values that don't have a parameter can't be transformed because that won't go through accept. 

I'll add a unit test for this case.